### ISSUE TITLE
Thoughts on point on arc

### ISSUE THREAD

**Issue by strobelm** [CindyJS/CindyJS#117]
Hi, i wanted to implement point on arc and @richter-gebert pointed out that if one moves the arc the cross ratio of the points should be preserved. I have drafted an algorithm which projects the four given points to l_inf and handles the problem like in RP^1:

```
// M is the center of the arc by A,B,C
linf = [0,0,1];

a = cross(linf, cross(A,M));
b = cross(linf, cross(B,M));
c = cross(linf, cross(C,M));
p = cross(linf, cross(P,M));


// cr
o = [0,0,1];
l = det(o,a,c)*det(o,b,p)/ (det(o,a,p)*det(o,b,c));

// given a crossratio l construct p such that [p,1,0] has crossratio l with given points a,b,c
// in the end: solve cr(a,b,c,p) for p = [p1, p2] with p2 = 1
p = a_1*b_1*c_2*l-a_1*b_1*c_2-a_1*b_2*c_1*l+a_2*b_1*c_1;
p = -p / (a_1*b_2*c_2 - a_2*b_1*c_2*l + a_2*b_2*c_1*l - a_2*b_2*c_1);

draw(line(cross(M, [p,1,0])));
// the desired point is now the intersection of the line and the arc.
```

This mimics exactly the behavior of Cinderella. The reason for the issue is that i'm not sure whether there is an easier way to construct the point. Any ideas?

**Comment by kranich** [comment]
You probably also want to consider the case that `[1,0,0]` has crossratio `l` with points `A`, `B`, `C`.

**Comment by gagern** [comment]
First of all, the center is an inappropriate concept here. Projecting from the center you get a double-cover of the line at infinity. Stereographic projection would be more appropriate, but I think I'd suggest something else.

The cross ratio of four points on a circle is the same as the cross ratio of these four points seen from a fifth point *on that circle* (which the center is not!). One convenient choice here would be the point *I*=[1,*i*,0] which lies on any circle and is usually not one of the defining points of an arc, as these tend to be real.

So you want to find *p* such that (a) cr(a,b;c,p)_I=const. and (b) *p* lies on the circle. (a) gives you a linear equation in the coordinates of *p*, which you can interpret as the homogeneous coordinates of a line. That line you have to intersect with the circle, but you know that the line will pass through *I*, so you can use the machinery behind OtherIntersection to obtain the point in question.

**Comment by strobelm** [comment]
Closed via #527.
